tests/test_uncrustify/failure.py

```diff
@@ -56,3 +56,15 @@ def __init__(self, expected, actual):
     def __str__(self):
         return 'Output {!r} does not match expected output {!r}'.format(
             self.actual_path, self.expected_path)
+
+# =============================================================================
+class UnexpectedlyPassing(Failure):
+    # -------------------------------------------------------------------------
+    def __init__(self, expected, actual):
+        self.expected_path = expected
+        self.actual_path = actual
+
+    # -------------------------------------------------------------------------
+    def __str__(self):
+        return 'Output {!r} was not expected to match expected output {!r}'.format(
```

Comment:
Pedantic: Per [PEP-8](https://www.python.org/dev/peps/pep-0008/#id19)¹ this line is too long (>79 chars). Please try:
```python
m = 'Output {!r} was not expected to match expected output {!r}'
return m.format(self.actual_path, self.expected_path)
```
(...which, conveniently, is the same number of lines. Feel free to use `msg` or whatever you prefer instead of `m`, so long as it fits.)

(¹ I try to mostly follow `pycodestyle` except for places where it is actively "detrimental", e.g. it forbids alignment of assignments.)

Reply:
Agree, we should enforce this with https://github.com/google/yapf or flake8 for instance and tell everyone how to install the linters locally to be able to check python formatting before pushing.

Reply:
I'm not familiar with yapf; I've always just used [pycodestyle](https://github.com/PyCQA/pycodestyle), but a) I'm ignoring E221, and b) I like separator lines (all the `# ===` and `# ---` comments), though I am using the latter in a way that pycodestyle tolerates.

At least one advantage of pycodestyle is that it is packaged on Fedora; yapf isn't. (Ubuntu seems to have both.) It's also dead simple to run: `pycodestyle tests/test_uncrustify/*.py`. (Or on any other Python files that are trying to be styled... Some of the old ones are a mess, but anything I've hacked on significantly should be conforming, aside from a couple E221 hits.)

I'd be okay with using flake8 if available; it uses pycodestyle under the hood, so it gives similar results, and folks that don't want to install all of it can use just pycodestyle.